### ISSUE TITLE
Question about test

### ISSUE THREAD

**Issue by zhqia** [shnhrtkyk/FWNet2#5]
I have run train.py, and got high acc, however got low acc or even precision zero in testing process by run infer,py, I have used the data with all labels and added weight. I don't know where is the problem.

**Comment by zhqia** [comment]
And I use test.txt for training, and also test in test.txt, but still got low precision.

**Comment by shnhrtkyk** [comment]
Did you use the same output from the model during the training process?
In the original infer.py, pred_batch = model(points) is not the same output of the model.

**Comment by zhqia** [comment]
> Did you use the same output from the model during the training process? In the original infer.py, pred_batch = model(points) is not the same output of the model.

Yes, the code in training process I used is: seg_both, seg_pred_local, seg_pred_global = model(points), and I use seg_both to get predict label. The code in testing process is: pred_batch, _, _ = model(points)

**Comment by zhqia** [comment]
And I am wondering the number of batchsize use in the code. the number of batchsize in paper is "1 for each GPU", so how many batchsize in total in the code ?

**Comment by shnhrtkyk** [comment]
We used four GPUs.  The total batch size is 4

**Comment by zhqia** [comment]
> We used four GPUs. The total batch size is 4

And do you use batch size 4 in testing process?

**Comment by zhqia** [comment]
And in train.py it seems like use only one batch. so how to read four batch?

**Comment by shnhrtkyk** [comment]
Only training process.
You can easiy choose the gpu id like this: https://pytorch.org/docs/stable/notes/cuda.html
And then, you should calulate the mean loss of individual loss value from each gpu.

**Comment by zhqia** [comment]
> Only training process. You can easiy choose the gpu id like this: https://pytorch.org/docs/stable/notes/cuda.html And then, you should calulate the mean loss of individual loss value from each gpu.

Thank you very much, Ihave learn that. I got a good result in training process, however got a much lower result in testing process, where do you think may be wrong?
The result in training and testing process is as follow:
Training :
Precision 0.99  0.99  1  0.99  0.97  0.98
 recall      0.99  0.99  1  0.99  0.99  0.99
 f1           0.99  0.99  1  0.99   0.98  0.98
Testing:
Precision 0.93  0.98  0.98  0.96  .088  0.92
recall       0.94  0.94  0.95  0.95  0.78  0.57
f1            0.82  0.96  0.96  0.96  0.83  0.71

**Comment by zhqia** [comment]
> Only training process. You can easiy choose the gpu id like this: https://pytorch.org/docs/stable/notes/cuda.html And then, you should calulate the mean loss of individual loss value from each gpu.

Thank you very much, Ihave learn that. I got a good result in training process, however got a much lower result in testing process, where do you think may be wrong?
The result in training and testing process is as follow:
Training :
Precision 0.99 0.99 1 0.99 0.97 0.98
recall 0.99 0.99 1 0.99 0.99 0.99
f1 0.99 0.99 1 0.99 0.98 0.98
Testing:
Precision 0.75 0.98 0.98 0.96 .088 0.92
recall 0.94 0.94 0.95 0.95 0.78 0.57
f1 0.82 0.96 0.96 0.96 0.83 0.71

**Comment by shnhrtkyk** [comment]
In this data, the Steet path is very difficult to classify.
And ground truth of test data has missed labels in the transmission tower class.
I think you have to re-classify the transmission tower class in the test data manually.

**Comment by zhqia** [comment]
> 
Thank you for your reply ! So what is the meaning of "re-classify the transmission tower class in the test data manually", does it means I should do a binary classification with the transmission tower and other category additionally?
